File: backend/app/api/tools.py
"""工具管理路由层（只做 HTTP 编排）。

契约主定义：specs/003-tool-management/contracts/api-contract.md
本模块只提供列表 / 详情 / 启停三个端点——系统内置工具不允许删除或修改用途
（FR-007），因此没有 DELETE 与其他 PUT 端点；统一执行入口为进程内服务
tool_executor（research R6），不经 HTTP 暴露。
"""

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.core.db import get_session
from app.schemas.tool import ToolDetail, ToolItem, ToolToggleRequest
from app.services import tool_service
from app.services.tool_service import ToolNotFoundError

router = APIRouter(prefix="/api/tools", tags=["tools"])


@router.get("", response_model=list[ToolItem])
def list_tools(session: Session = Depends(get_session)) -> object:
    return tool_service.list_tools(session)


@router.get("/{name}", response_model=ToolDetail)
def get_tool(name: str, session: Session = Depends(get_session)) -> object:
    try:
        return tool_service.get_tool(session, name)
    except ToolNotFoundError as exc:
        raise HTTPException(status_code=404, detail="工具不存在") from exc


@router.put("/{name}/enabled", response_model=ToolItem)
def set_tool_enabled(
    name: str,
    payload: ToolToggleRequest,
    session: Session = Depends(get_session),
) -> object:
    try:
        return tool_service.set_enabled(session, name, payload.enabled)
    except ToolNotFoundError as exc:
        raise HTTPException(status_code=404, detail="工具不存在") from exc

File: frontend/src/api/tools.ts
/**
 * 工具管理接口封装。
 *
 * 契约主定义：specs/003-tool-management/contracts/api-contract.md
 * 类型 MUST 与契约逐字段对齐（null 不混用 undefined）。
 */

import { http } from './request'

/** 工具参数定义（详情参数表的一行） */
export interface ToolParam {
  name: string
  type: 'string' | 'enum'
  required: boolean
  description: string
}

/** 列表项 */
export interface ToolItem {
  name: string
  display_name: string
  purpose: string
  params_summary: string
  enabled: boolean
  is_builtin: boolean
  updated_at: string
}

/** 详情（含参数表与三节说明） */
export interface ToolDetail extends ToolItem {
  params: ToolParam[]
  usage_scenarios: string
  input_requirements: string
  restrictions: string
}

/** 启停提交体 */
export interface ToolTogglePayload {
  enabled: boolean
}

export const toolsApi = {
  list(): Promise<ToolItem[]> {
    return http.get<ToolItem[]>('/tools')
  },
  detail(name: string): Promise<ToolDetail> {
    return http.get<ToolDetail>(`/tools/${name}`)
  },
  setEnabled(name: string, payload: ToolTogglePayload): Promise<ToolItem> {
    return http.put<ToolItem>(`/tools/${name}/enabled`, payload)
  },
}
